Neos.Media/Classes/Domain/Service/ThumbnailService.php

```diff
@@ -142,17 +142,16 @@ public function getThumbnail(AssetInterface $asset, ThumbnailConfiguration $conf
 
             // If the thumbnail strategy failed to generate a valid thumbnail
             if ($async === false && $thumbnail->getResource() === null && $thumbnail->getStaticResource() === null) {
+                // the thumbnail should not be persisted at this point, but remove is a no-op if the thumbnail
+                // does not exist - and if it does, this keeps it out of the way…
                 $this->thumbnailRepository->remove($thumbnail);
                 return null;
             }
 
             if (!$this->persistenceManager->isNewObject($asset)) {
```

Comment:
Does this actually do what it appears to be doing? See above note on the foreign key constraint error…